facebook/OpenBIC
# fby35: NF: Support sensor reading

Summary:
- Suport monitor temperature.
- Support monitor ADC voltage.
- Support monitor VR (xdpe12284c) voltage, current, and power.
- Support monitor INA233 voltage, current, and power.

Test plan:
- Build code: Pass
- Check sensor table: Pass

Note: Currently, we don't have NF board, sensor readings aren't verified.

Log:
uart:~$ platform sensor list_all
---------------------------------------------------------------------------------
[0x50] NF_1OU_BOARD_INLET_TEMP_C: tmp75      | access[O] | poll[O] 1    sec | 4byte_acur_read_success   |    24.000
[0x56] NF_ADC_P1V2_STBY_VOLT_V  : adc        | access[O] | poll[O] 1    sec | 4byte_acur_read_success   |     0.317
[0x57] NF_ADC_P1V2_ASIC_VOLT_V  : adc        | access[X] | poll[O] 1    sec | not_accesible             | na
[0x58] NF_ADC_P1V8_ASIC_VOLT_V  : adc        | access[X] | poll[O] 1    sec | not_accesible             | na
[0x59] NF_ADC_PVPP_AB_VOLT_V    : adc        | access[X] | poll[O] 1    sec | not_accesible             | na
[0x5a] NF_ADC_PVPP_CD_VOLT_V    : adc        | access[X] | poll[O] 1    sec | not_accesible             | na
[0x5b] NF_ADC_PVTT_AB_VOLT_V    : adc        | access[X] | poll[O] 1    sec | not_accesible             | na
[0x5c] NF_ADC_PVTT_CD_VOLT_V    : adc        | access[X] | poll[O] 1    sec | not_accesible             | na
[0x5d] NF_ADC_P0V75_ASIC_VOLT_V : adc        | access[X] | poll[O] 1    sec | not_accesible             | na
[0x5e] NF_INA233_P12V_STBY_VOLT_V: ina233     | access[O] | poll[O] 1    sec | 4byte_acur_read_success   |    12.463
[0x5f] NF_INA233_P3V3_STBY_VOLT_V: ina233     | access[O] | poll[O] 1    sec | 4byte_acur_read_success   |     3.318
[0x64] NF_INA233_P12V_STBY_CURR_A: ina233     | access[O] | poll[O] 1    sec | 4byte_acur_read_success   |     0.154
[0x65] NF_INA233_P3V3_STBY_CURR_A: ina233     | access[O] | poll[O] 1    sec | 4byte_acur_read_success   |     0.112
[0x6a] NF_INA233_P12V_STBY_PWR_W: ina233     | access[O] | poll[O] 1    sec | 4byte_acur_read_success   |     1.924
[0x6b] NF_INA233_P3V3_STBY_PWR_W: ina233     | access[O] | poll[O] 1    sec | 4byte_acur_read_success   |     0.375
[0x52] NF_VR_P0V85_ASIC_TEMP_C  : xdpe12284c | access[X] | poll[O] 1    sec | not_accesible             | na
[0x53] NF_VR_PVDDQ_AB_TEMP_C    : xdpe12284c | access[X] | poll[O] 1    sec | not_accesible             | na
[0x54] NF_VR_P0V8_ASIC_TEMP_C   : xdpe12284c | access[X] | poll[O] 1    sec | not_accesible             | na
[0x55] NF_VR_PVDDQ_CD_TEMP_C    : xdpe12284c | access[X] | poll[O] 1    sec | not_accesible             | na
[0x60] NF_VR_P0V85_ASIC_VOLT_V  : xdpe12284c | access[X] | poll[O] 1    sec | not_accesible             | na
[0x61] NF_VR_PVDDQ_AB_VOLT_V    : xdpe12284c | access[X] | poll[O] 1    sec | not_accesible             | na
[0x62] NF_VR_P0V8_ASIC_VOLT_V   : xdpe12284c | access[X] | poll[O] 1    sec | not_accesible             | na
[0x63] NF_VR_PVDDQ_CD_VOLT_V    : xdpe12284c | access[X] | poll[O] 1    sec | not_accesible             | na
[0x6c] NF_VR_P0V85_ASIC_PWR_W   : xdpe12284c | access[X] | poll[O] 1    sec | not_accesible             | na
[0x6d] NF_VR_PVDDQ_AB_PWR_W     : xdpe12284c | access[X] | poll[O] 1    sec | not_accesible             | na
[0x6e] NF_VR_P0V8_ASIC_PWR_W    : xdpe12284c | access[X] | poll[O] 1    sec | not_accesible             | na
[0x6f] NF_VR_PVDDQ_CD_PWR_W     : xdpe12284c | access[X] | poll[O] 1    sec | not_accesible             | na
[0x66] NF_VR_P0V85_ASIC_CURR_A  : xdpe12284c | access[X] | poll[O] 1    sec | not_accesible             | na
[0x67] NF_VR_PVDDQ_AB_CURR_A    : xdpe12284c | access[X] | poll[O] 1    sec | not_accesible             | na
[0x68] NF_VR_P0V8_ASIC_CURR_A   : xdpe12284c | access[X] | poll[O] 1    sec | not_accesible             | na
[0x69] NF_VR_PVDDQ_CD_CURR_A    : xdpe12284c | access[X] | poll[O] 1    sec | not_accesible             | na
---------------------------------------------------------------------------------
